src/staking/observable/index.ts

```diff
@@ -0,0 +1,372 @@
+   * 
+   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   */
+  public createWithdrawEarlyUnbondedTransaction = (
+    stakingParams: ObservableStakingParams,
+    delegation: ObservableDelegation,
+    unbondingTx: Transaction,
+    feeRate: number,
+  ): PsbtTransactionResult => {
+    validateParams(stakingParams);
+    const { stakingTx } = validateAndDecodeDelegationInputs(
+      delegation, stakingParams, this.stakerInfo,
+    );
+    // Build scripts
+    const scripts = buildScripts(
+      stakingParams,
+      delegation.finalityProviderPkNoCoordHex,
+      stakingTx.timelock,
+      delegation.stakerPkHex,
+    );
+
+    // Create the withdraw early unbonded transaction
+    try {
+      return withdrawEarlyUnbondedTransaction(
+        scripts,
+        unbondingTx,
+        this.stakerInfo.address,
+        this.network,
+        feeRate,
+      );  
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build unsigned withdraw early unbonded transaction",
+      );
+    }
+  }
+
+  /**
+   * Create a withdraw timelock unbonded transaction for observable staking.
+   * 
+   * @param {ObservableStakingParams} stakingParams - The staking parameters for observable staking.
+   * @param {ObservableDelegation} delegation - The delegation to withdraw from.
+   * @param {number} feeRate - The fee rate for the transaction in satoshis per byte.
+   * 
+   * @returns {PsbtTransactionResult} - An object containing the unsigned psbt and fee
+   * 
+   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   */
+  public createWithdrawTimelockUnbondedTransaction = (
+    stakingParams: ObservableStakingParams,
+    delegation: ObservableDelegation,
+    feeRate: number,
+  ): PsbtTransactionResult => {
+    validateParams(stakingParams);
+    const { stakingTx, btcStakingTx } = validateAndDecodeDelegationInputs(
+      delegation, stakingParams, this.stakerInfo,
+    );
+    // Build scripts
+    const scripts = buildScripts(
+      stakingParams,
+      delegation.finalityProviderPkNoCoordHex,
+      stakingTx.timelock,
+      delegation.stakerPkHex,
+    );
+
+    // Create the timelock unbonded transaction
+    try {
+      return withdrawTimelockUnbondedTransaction(
+        scripts,
+        btcStakingTx,
+        this.stakerInfo.address,
+        this.network,
+        feeRate,
+        stakingTx.stakingOutptuIndex,
+      );  
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build unsigned timelock unbonded transaction",
+      );
+    }
+  }
+}
+
+export const validateAndDecodeDelegationInputs = (
+  delegation: ObservableDelegation, stakingParams: ObservableStakingParams,
+  stakerInfo: StakerInfo,
+) => {
+  const stakingTx = delegation.stakingTx;
```

Comment:
What about the validation of the other fields of the delegation? For example, validating that the finality provider public key is valid key without a coordinate